src/dbm/dbm_multiply_comm.c

```diff
@@ -226,17 +227,17 @@ static void fill_send_buffers(
         for (int i = 0; i < row_size; i++) {
           for (int j = 0; j < col_size; j++) {
             const double element = blk_data[j * row_size + i];
-            norm += element * element;
             data_send[offset + i * col_size + j] = element;
+            norm += element * element;
           }
         }
         blks_send[jblock].free_index = plan->blk->col;
         blks_send[jblock].sum_index = plan->blk->row;
       } else {
-        for (int i = 0; i < row_size * col_size; i++) {
+        memcpy(data_send + offset, blk_data, sizeof(double) * plan_size);
```

Comment:
Is the `memcpy` really cheaper if we're looping over the same date again right afterwards? I would expect that the `element` could just remain in the register.

Reply:
No it's likely not cheaper. It actually looks nicer in the profiler now that all such zeroing tasks are summarized under the memset primitive. It's also not costlier since it is (hopefully) in-cache after memset and owned by the core in question. Also, memset is highly optimized in GLIBC; just in case...

Reply:
... but I can change it back if you prefer.

Reply:
Reverted back to in-loop update.